mmv1/products/netapp/storagePool.yaml

```diff
@@ -0,0 +1,228 @@
+  # If false (or unset), the provider sets the resource's Terraform ID before the resource is
+  # created, based only on the resource configuration.
+  # result: !ruby/object:Api::OpAsync::Result
+  #   resource_inside_response: true
+
+# All resources (of all kinds) that share a mutex value block rather than
+# executing concurrent API requests.
+# Terraform field names enclosed in double curly braces are replaced with
+# the field values from the resource at runtime.
+# mutex: RESOURCE_NAME/{{name}}
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      LOCATION_DESCRIPTION
+  - !ruby/object:Api::Type::String
+    name: 'storage_pool_id'
+    description:
+      'The unique name of the storage pool
+      `[_a-zA-Z0-9][-_.a-zA-Z0-9]*`.'
+    required: true
+    immutable: true
+    url_param_only: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'Storage_pool_create'
+    primary_resource_id: 'test_pool'
+    vars:
+      pool_name: 'test-pool'
+properties:
+  # Fields go here
+  - !ruby/object:Api::Type::String
+    name: name
+    output: true
+    description: |
+      Name of the storage pool
+  - !ruby/object:Api::Type::Enum
+    name: 'serviceLevel'
+    description: |
+      Service level of the storage pool
+    values:
+      - :PREMIUM
+      - :EXTREME
+      - :STANDARD
+  - !ruby/object:Api::Type::String
+    name: 'capacityGib'
+    description: |
+      Capacity in GIB of the pool
+  - !ruby/object:Api::Type::String
+    name: 'volumeCapacityGib'
+    description: |
+      Allocated size of all volumes in GIB in the storage pool
+    output: true
+  - !ruby/object:Api::Type::Integer
+    name: 'volumeCount'
+    description: |
+      Volume count of the storage pool
+    output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'State'
+    description: |
+      State of the storage pool
+    values:
+      - :STATE_UNSPECIFIED
+      - :CREATING
+      - :RESTORING
+      - :UPDATING
+      - :DELETING
+      - :DISABLED
+      - :READY
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'stateDetails'
+    description: |
+      State details of the storage pool
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'createTime'
+    description: |
+      Create time of the storage pool
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      Description of the storage pool
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: labels
+    description: |
+      Labels as key value pairs.An object containing a list of "key": value pairs. Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.
+  - !ruby/object:Api::Type::String
+    name: 'network'
+    description: |
+      VPC Network name. Format: projects/{project}/global/networks/{network}
+    diff_suppress_func: tpgresource.ProjectNumberDiffSuppress
```

Comment:
The API change is still in process. it's yet to go into production as per my last conversation with Jesus.

Reply:
Thanks for the update! Do we have an estimation on when the API change will be rolled out fully? Plus, what it's going to be? Accepting both project number and project id and return whatever is sent to the API or only accepting project number as the valid value?

Reply:
@shuyama1 I've pasted the query in google chat group to Jesus. Is this a blocker for us to get storage pool merged?

Reply:
Thanks @G-NamanGupta! I'll wait Jesus to confirm the expected behavior for this field and when the change will be rolled out. My concern here is that I remember the API will be changed to only accept project number as Jesus mentioned before, but in our example, it's still using `google_compute_network.peering_network.id` which uses the project id. If we release the resource now and when the API change is rolled out, it will break our tests as well as users who uses project id for it, I believe. Sorry for being a little extra cautious here.

Reply:
Got confirmed from Jesus that project number will only be accepted after the API change is rolled out. We'll need to update all the tests/examples to use only project number even if sending project id is also working now. This is to avoid test fails after the API change is out or generating incorrect configuration example to users that will eventually break themselves. Thank you for working on it and apologize for the extra work.